library/core/src/char/convert.rs

```diff
@@ -50,8 +50,12 @@ use super::MAX;
 #[doc(alias = "chr")]
 #[inline]
 #[stable(feature = "rust1", since = "1.0.0")]
-pub fn from_u32(i: u32) -> Option<char> {
-    char::try_from(i).ok()
+#[rustc_const_unstable(feature = "const_char_convert", issue = "89259")]
+pub const fn from_u32(i: u32) -> Option<char> {
+    match char_try_from_u32(i) {
```

Comment:
I wish we wouldn't have to sacrifice readability for constability, especially because we can eventually have both when `Result::ok` is `const` too. How about adding a `FIXME` or similar to change this back later?

Reply:
I've tried making `Result::ok` const, but it is blocked by const drop.

Reply:
I suggested to add a `FIXME`, not to fix it immediately.